plugins/modules/zos_job_submit.py

```diff
@@ -902,7 +899,7 @@ def run_module():
     from_encoding = parsed_args.get("from_encoding")
     to_encoding = parsed_args.get("to_encoding")
     # temporary file names for copied files when user sets location to LOCAL
-    temp_file = parsed_args.get("temp_file")
+    temp_file = parsed_args.get("src")
```

Comment:
I would suggest doing this change instead : 

```suggestion
    if location == "LOCAL":
        temp_file = parsed_args.get("src")
```

That way `temp_file` is indeed a temporary file on the remote and we don't need to worry on the `location` conditions in the future, if `temp_file` is None we know we don't have to remove anything.